### PR TITLE
53 add dash abilities and 32 be able to push creeps out of the way

### PR DESCRIPTION
**Pull request by mitchellkazin15**
To dash, press space or A on xbox controller. You start with a basic dash that gains abilities as you level up
level 2: gain brief invulnerability
level 3: gain small damage AOE while dashing
level 4: gain AOE knockback effect
level 5: gain intangibility and dash through enemies

Minor addition you can also push enemies around so you can escape being surrounded

### CODE REVIEW

File: characters/displaceable_character_body.gd [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		displaced_timer = get_tree().create_timer(stat_component.get_current_displaced_duration_sec())
+	displaced = true
+
+
+func accelerate_and_collide(delta):
+	if displaced_timer and displaced_timer.time_left == 0.0:
+		tmp_max_speed_mult = base_displaced_max_speed_mult
+		displaced = false
+	if displaced:
+		print()
```
Comment (wchaney98) [inline characters/displaceable_character_body.gd:47]: Rogue print statement

File: characters/displaceable_character_body.gd [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		var tmp_move_dir = move_direction if move_direction != Vector2.ZERO else -velocity.normalized()
```
Comment (wchaney98) [inline characters/displaceable_character_body.gd:48]: Unused var? Should this be used on the next line?

**Review by wchaney98 — COMMENTED** [review]
Other comments:
* Like we all talked about, the dash displacement of enemies feels and looks a bit off. Often it feels like just dashing through the enemies (without collision) would be more beneficial. I think it would be worth exploring alternatives
* The dash icon covers the bottom gun, and also doesn't indicate what your dash actually does (since there's multiple levels to it). Maybe a new ticket to figure out a better UI for that?

Otherwise LGTM, pretty fun to play and test

**Comment by mitchellkazin15** [comment]
> Other comments:
> 
> * Like we all talked about, the dash displacement of enemies feels and looks a bit off. Often it feels like just dashing through the enemies (without collision) would be more beneficial. I think it would be worth exploring alternatives
> * The dash icon covers the bottom gun, and also doesn't indicate what your dash actually does (since there's multiple levels to it). Maybe a new ticket to figure out a better UI for that?
> 
> Otherwise LGTM, pretty fun to play and test

Thanks for the review! Ended up removing the knockback effect completely from the dash. If it needs reworking I can do that when I work on other abilities. And then I moved the dash cooldown icon to the HUD

File: characters/displaceable_character_body.gd (hunk 2 above)
Comment (mitchellkazin15) [reply characters/displaceable_character_body.gd]: Oh yep, leftover artefact from when I was throwing everything at the wall. removed

**Review by wchaney98 — APPROVED** [review]

**Review by ninja-rage — APPROVED** [review]

**Pull request merged**